kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/ConversationUIService.java

```diff
@@ -646,23 +644,21 @@ public void updateAgentStatus(String userId, Integer status) {
-                if(status.equals(KmConstants.STATUS_AWAY)) {
-                    getConversationFragment().switchContactStatus(baseContactService.getContactById(userId), false);
-                    getConversationFragment().showAwayMessage(true, null);
-                } else if(status.equals(KmConstants.STATUS_ONLINE)) {
-                    getConversationFragment().switchContactStatus(baseContactService.getContactById(userId), true);
-                    getConversationFragment().showAwayMessage(false, null);
-                }
-                else if(status.equals(KmConstants.STATUS_OFFLINE)){
-                    getConversationFragment().processSupportGroupDetails(getConversationFragment().getChannel());
-                    getConversationFragment().showAwayMessage(true, null);
-                }
-                else if(status.equals(KmConstants.STATUS_CONNECTED)){
-                    getConversationFragment().processSupportGroupDetails(getConversationFragment().getChannel());
-                    getConversationFragment().loadAwayMessage();
-                }
-        } else if(userId != null && status != null && getConversationFragment().getChannel() != null && KmChannelService.getInstance(fragmentActivity).getUserInSupportGroup(getConversationFragment().getChannel().getKey()).equals(userId)) {
+        if (userId != null && status != null && !KmUtils.isAgent() && getConversationFragment().getChannel() != null && !TextUtils.isEmpty(getConversationFragment().getChannel().getConversationAssignee()) && getConversationFragment().getChannel().getConversationAssignee().equals(userId)) {
+            if (status.equals(KmConstants.STATUS_AWAY)) {
+                getConversationFragment().switchContactStatus(baseContactService.getContactById(userId), false);
+                getConversationFragment().showAwayMessage(true, null);
+            } else if (status.equals(KmConstants.STATUS_ONLINE)) {
+                getConversationFragment().switchContactStatus(baseContactService.getContactById(userId), true);
+                getConversationFragment().showAwayMessage(false, null);
+            } else if (status.equals(KmConstants.STATUS_OFFLINE)) {
+                getConversationFragment().processSupportGroupDetails(getConversationFragment().getChannel());
+                getConversationFragment().showAwayMessage(true, null);
+            } else if (status.equals(KmConstants.STATUS_CONNECTED)) {
+                getConversationFragment().processSupportGroupDetails(getConversationFragment().getChannel());
+                getConversationFragment().loadAwayMessage();
+            }
+        } else if (userId != null && status != null && getConversationFragment().getChannel() != null && KmChannelService.getInstance(fragmentActivity).getUserInSupportGroup(getConversationFragment().getChannel().getKey()).equals(userId)) {
```

Comment:
Not sure why these are showing as changes in Github, make sure not to change existing lines.

Reply:
When pushing the changes, try to avoid pushing unrequired changes, it makes review difficult.

Reply:
i reformat the code thats why its showing that